lib/mixlib/install/cli.rb

```diff
@@ -75,14 +75,15 @@ def download(product_name)
         if options[:url]
           say artifact.url
         else
+          url = artifact.is_a?(Array) ? artifact.map(&:url).first : artifact.url
```

Comment:
`Mixlib::Install.detect_platform` => `{:platform=>nil, :platform_version=>nil, :architecture=>nil}` which leads artifact to be an array containing an artifact for 51 platform/platform_version combinations.

This is because Mixlib::Install.detect_platform doesn't validate the results of Mixlib::ShellOut. In one case, powershell.exe doesn't exist and the command fails, but we still put empty values in the hash we return. #188 was apparently against Windows Server 2008sp2, which doesn't have powershell installed by default and when it does you get powershell v1.0 which doesn't take the `-file` option. `type platform_detection.ps1 | powershell -command -` errors too. We should probably use #188 to return an error when Mixlib::ShellOut fails, but @mwrock has reminded me that making these powershell scripts compatible with powershell 1.0 is pretty untenable.